languagetool-core/src/main/java/org/languagetool/tagging/disambiguation/AbstractDisambiguator.java

```diff
@@ -32,4 +42,84 @@ public AnalyzedSentence preDisambiguate(AnalyzedSentence input) {
     return input;
   }
 
+  private SpellingCheckRule spellingRule = null;
+  private List<String> commonWords = new ArrayList<>();
+  private final List<String> commonEnglishWords = Arrays.asList("the", "to", "of", "and", "in", "a", "was", "is",
+    "this", "i", "as", "for", "on", "with", "by", "that", "from", "at", "his", "an", "he", "are", "were", "don't",
+    "which", "it", "or", "had", "has", "one", "but", "first", "very", "we", "not", "their", "any", "have", "who",
+    "been", "her", "two", "get", "other", "after", "into", "they", "time", "all", "be", "me", "more", "must", "only",
+    "when", "years", "most", "over", "used", "would", "you", "can", "during", "going", "good", "him", "new", "out",
+    "she", "such", "where", "between", "made", "many", "some", "then", "there", "year", "about", "back", "became",
+    "later", "part", "under", "well", "being", "including", "them", "through", "area", "both", "make", "name",
+    "before", "called", "could", "fine", "people", "second", "until", "while", "will", "i'll", "against", "age",
+    "city", "go", "may", "number", "population", "season", "several", "team", "these", "work", "born", "early",
+    "family", "film", "now", "same", "series", "so", "use", "what", "album", "based", "four", "it's", "life",
+    "released", "since", "state", "began", "century", "each", "end", "following", "found", "game", "high", "located",
+    "town", "i've", "around", "because", "can't", "come", "day", "did", "didn't", "government", "group", "here",
+    "home", "large", "like", "love", "music", "my", "named", "no", "often", "system", "that's", "three", "too", "up",
+    "us", "won", "yes", "you're", "along", "built", "career", "former", "if", "include", "left", "line", "local",
+    "long", "off", "own", "place", "small", "still", "those", "took", "band", "due", "form", "held", "its", "last",
+    "major", "member", "members", "much", "don", "didn", "doesn", "wasn", "weren", "'s", "'t", "'ll");
```

Comment:
The one thing I think we will need is language-specific lists of words (ideally kept in txts that are then read by the disambiguator), as some of these words are also valid words in other languages and shouldn't be used to make judgements about Englishiness.

Reply:
Right, I think I understand the idea. I've pulled this locally and added some Portuguese tests, and I think my scepticism about having common Portuguese function words annotated as 'common English' words is justified:

```
# Correct
Sentence: "Um bom film." // Matches: 1
[filme, fim, fila, filo, file, filma, filé, filmá, filem, filó, filá, pium, FILs, Pium, fiam, fiem, filam, filmo, mium]

# Incorrect; I expect 'film' to be corrected, but because it is between very common words 'no' and 'as', it is skipped
Sentence: "No film as meninax discutem assuntos." // Matches: 1
[menina, meninas, minimax, meninão, memina, meminas, meminha, meminhas, meninez]
```

---

Furthermore, taking the examples above, the whole point of my original idea was to then run the spellchecking for **English**, but I'm getting Portuguese suggestions. The idea was that, when in close proximity to English words, incorrect words are also probably in English. E.g.:

```
Sentence: "Ele gosta de The Ride of the Valkiries" // Matches: 1
[Valquírias]
```

Here, since the whole thing is in English, I want `Valkyries`. `The Ride of the Valquírias` makes no sense.

Reply:
I see. 
Obviously, we can make adjustments to this method according to the needs of each language. Anything can be overwritten. Or you can use something entirely different. 

If you want to provide English spelling suggestions, you have to do it in the Portuguese spelling rule. This will be more language-dependent. Now, you know how to access the English speller rule without adding a dependency in pom.xml.

I am going to merge this for ES and CA. I want to see the nightly diffs and what happens in production.

Reply:
Fair, I'm curious to see what happens. We will prob. have to use a different list of English words and work on the suggestions.